src/storage/kv/transaction.rs

```diff
@@ -796,6 +800,55 @@ impl Transaction {
             Err(e) => Err(e),
         }
     }
+
+    /// Scans a range of keys and returns a vector of tuples containing the key, value, timestamp, and deletion status for each key.
+    pub fn scan_all_versions<'b, R>(
+        &self,
+        range: R,
+        limit: Option<usize>,
+    ) -> Result<Vec<ScanVersionResult>>
+    where
+        R: RangeBounds<&'b [u8]>,
+    {
+        // Convert the range to a tuple of bounds of variable keys.
+        let range = convert_range_bounds(range);
+
+        // Initialize an empty vector to store the results.
+        let mut results = Vec::new();
+
+        // Get a range iterator for the specified range.
+        let snap = self.snapshot.as_ref().unwrap().read();
+        let ranger = snap.range_with_versions(range);
+
+        // Iterate over the keys in the range.
+        for (key, value, _, ts) in ranger {
+            // If a limit is set and we've already got enough results, break the loop.
+            if let Some(limit) = limit {
+                if results.len() >= limit {
+                    break;
+                }
+            }
+
+            // Determine if the record is soft deleted based on the metadata.
+            let mut is_deleted = false;
```

Comment:
This can be simplified with the `Option::is_some_and` predicate
```rust
let is_deleted = value.metadata().is_some_and(|md| md.is_tombstone());
```

Reply:
Learnt something new! Fixed